### PR TITLE
documents: improve editor layout

### PR DESCRIPTION
**Pull request by jma**
* Improves document JSONSchema form options by adding css classes,
default values and form options to increase his usability.
* Makes the provision activity field required.
* Fixes issuance marc21 to JSON conversion.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1542?milestone=269576

## Dependencies

* https://github.com/rero/ng-core/pull/227
* https://github.com/rero/rero-ils-ui/pull/325

## How to test?

- Open the document editor and see how well it is.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/documents/jsonschemas/documents/document_issuance-v0.0.1.json [hunk 1]
```diff
@@ -242,7 +242,7 @@
     ],
     "form": {
       "templateOptions": {
-        "cssClass": "rero-ils-editor-title"
+        "cssClass": "editor-title bla"
```
Comment (AoNoOokami) [inline rero_ils/modules/documents/jsonschemas/documents/document_issuance-v0.0.1.json:245]: Remove "bla"

**Comment by PascalRepond** [comment]
- Default fields are added when editing existing document or importing but they should only appear when creating new document (open issue or other US)
- provisionActivity : startDate and endDate display problem (in Firefox) : field is too small
image
- Firefox - shortcut links still seem inconsistent
    - When I click a shortcut, where it scrolls depends on its position when I click : 
     --- If it was below the viewport, the first input field of the group goes to the bottom of the viewport
     --- If it was above, the first input field of the group goes to the top of the viewport
     --- If it was already in the viewport, the top-bar of the group goes to the top of the viewport (expected behavior)
- Languages and translations are correctly updated when changing language on-the-fly except  in the "Add fields" arrow dropdown : 
image

**Comment by JoelleDosimont** [comment]
Not much to add ;-)
- Mode of issuance resets to single unit, as Nicolas mentionned. It's the same with Author when it's an Organisation, it resets to Person.

**Comment by jma** [comment]
All the problems discover by Nicolas and Pascal has been addressed except:
- two fields: document type and extent do not have header. An issue has been created by Nicolas.
- the default values are not set when a field becomes visible, An issue has been created by Nicolas.


Notes:
- the provision activity is now required.
- the layout should be ok for screen bigger than 1400px wide
- new line is allowed for labels
- only the required fields are displayed for a new document additional fields, according to Nicolas  other fields can be displayed with the use of templates.

**Pull request merged**